packages/arb-token-bridge-ui/src/hooks/useClaimWithdrawal.ts

```diff
@@ -15,7 +20,19 @@ export function useClaimWithdrawal(): UseClaimWithdrawalResult {
   const {
     app: { arbTokenBridge }
   } = useAppState()
-  const { data: signer } = useSigner()
+  const {
+    l1: { network: l1Network },
+    isConnectedToArbitrum
+  } = useNetworksAndSigners()
+  const { data: l1Signer } = useSigner({
+    chainId: l1Network.chainID
+  })
+  const { switchNetworkAsync } = useSwitchNetwork({
+    throwForSwitchChainNotSupported: true,
+    onMutate: () =>
+      handleSwitchNetworkOnMutate({ isSwitchingNetworkBeforeTx: true }),
+    onError: handleSwitchNetworkError
+  })
```

Comment:
Not for this PR, but I'm seeing we use `useSwitchNetwork` a bunch of time with the same props. Maybe worth extracting to a custom hook?

Reply:
yea custom hook sounds fair